src/commands/weather/command.ts

```diff
@@ -29,48 +29,52 @@ const command: Command = {
   enabled: true,
   usage: '[invocation] [city | state | zip | etc]',
   async execute(message, args) {
-    const userLocation = await getUserLocation(message.author.username, args, message);
-    if (!userLocation) {
-      message.channel.send('Set your default location with .weather set YOUR_LOCATION');
-    }
-
     try {
-      const geoData = await getGeoLocation(userLocation);
-      if (!geoData) {
-        message.channel.send('Location was not found!', { files: [await getRandomEmotePath()] });
+      let user = await DI.userRepository.findOne(message.author.id);
+      const isUpdatingLocation = args[0]?.toLowerCase().trim() === 'set';
+      const location = await getUserLocation(user, args, isUpdatingLocation);
```

Comment:
I feel like this `getUserLocation` is doing work it shouldn't be doing inside of it. Like parsing additional args. Perhaps it shouldn't even exist.

You can use the main body of the `execute` method to process subcommands like done here:
https://github.com/BrycePearce/KyuuBot/blob/main/src/commands/reminder/command.ts#L64

Reply:
Not finding a great way to do this without duplicating code

Reply:
refactored it a bit